sdk/src/client/stronghold/mod.rs

```diff
@@ -149,7 +149,10 @@ fn check_or_create_snapshot(
 /// Extra / custom builder method implementations.
 impl StrongholdAdapterBuilder {
     /// Use an user-input password string to derive a key to use Stronghold.
-    pub fn password(mut self, password: &str) -> Self {
+    pub fn password(mut self, password: impl Into<Password>) -> Self {
+        // Note: we deliberately put the password conversion right here. Only change if you know what you're doing.
```

Comment:
This is one of those comments that is both critically important and profoundly unhelpful.

Reply:
agreed

Reply:
I always hate those kind of comments which is why I would have preferred the slight inconvenience in case the special case of zeroable passwords/mnemonics. Just doesn't feel right to me to rely on no one accidentally moving this code.

Reply:
the comments were supposed to prevent changes like this in the future:
```
let mut key_provider_guard = self.key_provider.lock().await?;
let key_provider = self::common::key_provider_from_password(password.into());
```
but yeah, maybe a bit over-cautious on my side.

Reply:
removed all those comments